runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -202,7 +202,16 @@ protected ILogData getNextEntry(QueuedStreamContext context,
      *
      * @param address       The address to read.
      */
-    protected abstract @Nonnull ILogData read(final long address);
+    protected @Nonnull ILogData read(final long address) {
+        try {
+            return readImpl(address);
```

Comment:
Agree with Medhavi, both 2 and 3 are viable, since we really only have one implementation for now; slight preference to 3 as well.